test/TestCase.cpp

```diff
@@ -139,3 +138,36 @@ EVMVersionRestrictedTestCase::EVMVersionRestrictedTestCase(std::string const& _f
 	if (!comparisonResult)
 		m_shouldRun = false;
 }
+
+void EVMVersionRestrictedTestCase::processCompileToFlag()
+{
+	std::optional<uint8_t> eofVersion = solidity::test::CommonOptions::get().eofVersion();
+
+	std::string compileToString = m_reader.stringSetting("compileTo", "any");
+	if (compileToString == "any")
+		return;
+
+	// TODO: This is naive implementation because for now we support only one EOF version.
+	if (compileToString == "legacy" && eofVersion.has_value())
+		m_shouldRun = false;
+	else if (compileToString == ">=1")
+	{
+		// Do not run if non-EOF test run
+		if (!eofVersion.has_value())
+			m_shouldRun = false;
+		// Do not run if before prague fork test run
+		else if (solidity::test::CommonOptions::get().evmVersion() < langutil::EVMVersion::prague())
```

Comment:
We should actually just refuse to run with this combination, since it makes no sense. It could be a validation error in [`CommonOptions::validate()`](https://github.com/ethereum/solidity/blob/v0.8.27/test/Common.cpp#L124-L158) and then here it would just be an assert.

Reply:
Actually, do we even reject that combination in `solc` CLI and Standard JSON? Because I think we should.

This is something we could do in a separate PR though. Or maybe just add it to #15310 to do it later, not to get too deep into distractions here :)

Reply:
Assert added, but in case when already`m_shouldRun == false` we need to skip checking eof version otherwise it asserts.

Reply:
But wasn't it actually because https://github.com/ethereum/solidity/pull/15471#discussion_r1792514663?

Reply:
right